### ISSUE TITLE
Parallellize update_beta

### ISSUE THREAD

**Issue by rfhh** [ielhelw/MCMC_FOR_AMMSB#23]
When the minibatch size grows, update_beta becomes a candidate for parallellization. My guess is that that is straightforward: calculate the grads in parallel, sum them globally (much like perplexity).

**Comment by rfhh** [comment]
Done, committed, merged into master.
